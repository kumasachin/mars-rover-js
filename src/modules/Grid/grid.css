.grid-column {
    min-width: 30px;
    min-height: 40px;
}
.grid-column span {
    min-height: 30px;
    display: block;
}
.grid-row {
    min-height: 40px;
}

.lostRow .lostCell {
    background-color: bisque;
}