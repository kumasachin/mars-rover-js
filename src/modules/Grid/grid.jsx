import React, { useState, useEffect } from 'react';
import { GridRow } from '../';
import './grid.css';

export const Grid = () => {
  return (
    <table
      cellSpacing="0"
      border="1"
      width="100%"
      height="100%"
      className="grid-container"
    >
      <tbody>
        <GridRow />
      </tbody>
    </table>
  );
};

export default Grid;
