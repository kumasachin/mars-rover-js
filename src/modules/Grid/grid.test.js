import { render, screen, cleanup } from '@testing-library/react';
import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Grid from './Grid';

Enzyme.configure({ adapter: new Adapter() });

const data = [
  {
    name: 'R1',
    color: 'green',
    currentPosition: '0 0 N',
    grid: '50 50',
    x: 0,
    y: 1,
    d: 'N',
    instructions: 'FFFFFFFFFFFFFFFFFFF',
  },
  {
    name: 'R2',
    color: 'green',
    grid: '3 3',
    currentPosition: '0 14 N',
    x: 0,
    y: 1,
    d: 'N',
    instructions: 'FRFFFFFFF',
  },
];

jest.mock('../../context/marsContext', () => {
  return {
    useMarsContextAPI: () => data,
  };
});
describe('Grid row snapshot', () => {
  afterEach(cleanup);

  it('should take a snapshot', () => {
    const { asFragment } = render(<Grid />);
    expect(asFragment(<Grid />)).toMatchSnapshot();
  });
});
