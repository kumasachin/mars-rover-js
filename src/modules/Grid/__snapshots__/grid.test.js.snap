// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid row snapshot should take a snapshot 1`] = `
<DocumentFragment>
  <table
    border="1"
    cellspacing="0"
    class="grid-container"
    height="100%"
    width="100%"
  >
    <tbody>
      <tr
        class="grid-row row-0"
      >
        <td
          class="grid-column column-0"
        >
          <span />
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
